feat: add brazilian portuguese language

thedmmatt opened the pull request:
Pull request from `ptocular` branch to Ocular's `main` branch to provide Portuguese (Brazil) as a language option.

Both [Contributing Guidelines](https://github.com/simonwep/ocular/blob/e99e8f1847f3d034ac99b073d4568d72ca4091e2/.github/CONTRIBUTING.md) and ["turkish" pull request](https://github.com/simonwep/ocular/pull/40#pullrequestreview-2051315183) have been followed to the extent of all guidance provided, except by `test:ci` routine which did not respond to move `build` as a PR. Please advise if additional steps are required.

thedmmatt commented:
@simonwep, I'm double-checking the `pt-br.json` file for any issues, but so far the `CI / build` routine yet throws an error from the ESLint rules, specifically at the file `Draggable.vue` which I'm still looking into.

The weird part is that GH suggests ln. 130 col. 7 as the culprit, but... it's just CSS styling. So, maybe reviewing the CI/CD pipeline or maybe skipping Vue's attr-orders altogether?

simonwep commented:
> @simonwep, I'm double-checking the `pt-br.json` file for any issues, but so far the `CI / build` routine yet throws an error from the ESLint rules, specifically at the file `Draggable.vue` which I'm still looking into.
> 
> The weird part is that GH suggests ln. 130 col. 7 as the culprit, but... it's just CSS styling. So, maybe reviewing the CI/CD pipeline or maybe skipping Vue's attr-orders altogether?
> 
> Edit 1: Debug just shows that the issue is an invalid character `\"`, which it's not present in `Draggable.vue` (since I did not edit it), but of course, is present in `pt-br.json` to escape double quotes. It should still work, but I'll try removing them if the next CI workflow does not work.
> 
> Edit 2: Just fixed a syntax error in `pt-br.json` where `page > header` had a missing comma. If this is the culprit, GH was pointing to the wrong file and line.

No worries :) I changed and force-pushed a few things, since I'm currently setting up contributing guidelines and tinkering what others need to successfully contribute:
- There was a typo in your localization file, `unknownError` vs `unkownError` - yeah, the TS error message is pretty crappy.
- Please use conventional commits, as already mentioned in the contributing guidelines.
- I squashed all your commits since fixes usually don't need an extra commit.

In the coming days I'll merge this one, thank you so much for your contribution ❤️

thedmmatt commented:
> > @simonwep, I'm double-checking the `pt-br.json` file for any issues, but so far the `CI / build` routine yet throws an error from the ESLint rules, specifically at the file `Draggable.vue` which I'm still looking into.
> > The weird part is that GH suggests ln. 130 col. 7 as the culprit, but... it's just CSS styling. So, maybe reviewing the CI/CD pipeline or maybe skipping Vue's attr-orders altogether?
> > Edit 1: Debug just shows that the issue is an invalid character `\"`, which it's not present in `Draggable.vue` (since I did not edit it), but of course, is present in `pt-br.json` to escape double quotes. It should still work, but I'll try removing them if the next CI workflow does not work.
> > Edit 2: Just fixed a syntax error in `pt-br.json` where `page > header` had a missing comma. If this is the culprit, GH was pointing to the wrong file and line.
> 
> No worries :) I changed and force-pushed a few things, since I'm currently setting up contributing guidelines and tinkering what others need to successfully contribute:
> 
> * There was a typo in your localization file, `unknownError` vs `unkownError` - yeah, the TS error message is pretty crappy.
> * Please use conventional commits, as already mentioned in the contributing guidelines.
> * I squashed all your commits since fixes usually don't need an extra commit.
> 
> In the coming days I'll merge this one, thank you so much for your contribution ❤️

Immediately after the CI success notification from GH, I noticed the stupid `br.json` calling in `locales` where the file was actually named `pt-br.json`. 😅 Thank you for spotting that, and I'm glad to help! 🤩
